dbeam-core/src/main/scala/com/spotify/dbeam/JdbcAvroRecord.java

```diff
@@ -84,42 +86,47 @@ private static ByteBuffer nullableBytes(byte[] bts) {
     }
   }
 
-  static SQLFunction<ResultSet, Object> computeMapping(ResultSetMetaData meta, int column)
+  static SQLFunction<ResultSet, Object> computeMapping(final ResultSetMetaData meta, final int column)
       throws SQLException {
-    final int columnType = meta.getColumnType(column);
-    if (columnType == VARCHAR || columnType == CHAR ||
-        columnType == CLOB || columnType == LONGNVARCHAR ||
-        columnType == LONGVARCHAR || columnType == NCHAR) {
-      return resultSet -> resultSet.getString(column);
-    } else if (columnType == BIGINT) {
-      final int precision = meta.getPrecision(column);
-      if (precision > 0 && precision <= MAX_DIGITS_BIGINT) {
-        return resultSet -> resultSet.getLong(column);
-      }
-      // otherwise return as string
-    } else if (columnType == INTEGER || columnType == SMALLINT || columnType == TINYINT) {
-      return resultSet -> resultSet.getInt(column);
-    } else if (columnType == TIMESTAMP || columnType == DATE ||
-               columnType == TIME || columnType == TIME_WITH_TIMEZONE) {
-      return resultSet -> {
-        final Timestamp timestamp = resultSet.getTimestamp(column, CALENDAR);
-        if (timestamp != null) {
-          return timestamp.getTime();
+    switch (meta.getColumnType(column)) {
+      case VARCHAR: case CHAR: case CLOB:
+      case LONGNVARCHAR: case LONGVARCHAR: case NCHAR:
+        return resultSet -> resultSet.getString(column);
+      case BIGINT:
+        final int precision = meta.getPrecision(column);
+        if (precision > 0 && precision <= MAX_DIGITS_BIGINT) {
+          return resultSet -> resultSet.getLong(column);
+        }
+        // otherwise return as string
+        break;
+      case INTEGER: case SMALLINT: case TINYINT:
+        return resultSet -> resultSet.getInt(column);
+      case TIMESTAMP: case DATE:
+      case TIME: case TIME_WITH_TIMEZONE:
+        return resultSet -> {
+          final Timestamp timestamp = resultSet.getTimestamp(column, CALENDAR);
+          if (timestamp != null) {
+            return timestamp.getTime();
+          } else {
+            return null;
```

Comment:
Can we avoid this null in any way?

Reply:
Columns in database can be nullable, DBeam handles data by making all fields from the avro record nullable. About Timestamp, on Avro we expose milliseconds since epoch, the only way to get that from JDBC Result set is using `getTimestamp(column).getTime()`. Hence we need to have this logic.

Reply:
okay 👍